lib/activerecord_spanner_adapter/connection.rb

```diff
@@ -44,11 +44,20 @@ def self.information_schema config
     end
 
     def session
+      @last_used = Time.current
       @session ||= spanner.create_session instance_id, database_id
     end
     alias connect! session
 
     def active?
+      # This method should not initialize a session.
+      unless @session
+        return false
+      end
+      # Assume that it is still active if it has been used in the past 50 minutes.
+      if ((Time.current - @last_used) / 60).round < 50
```

Comment:
This prevents the connection pool from executing a `SELECT 1` query __each__ time a connection is checked out of the pool. This check should be safe enough to return the correct answer in by far the most cases. Functions that rely on the session (queries, updates, ...) should take into account that a session could be invalidated by the backend, even when this time has not been exceeded.